contracts/modules/cover/CoverProducts.sol

```diff
@@ -214,16 +214,15 @@ contract CoverProducts is ICoverProducts, MasterAwareV2, Multicall {
         revert CapacityReductionRatioAbove100Percent();
       }
 
-      // TODO: https://github.com/NexusMutual/smart-contracts/issues/859
-      if (product.useFixedPrice) {
-        uint productId = param.productId == type(uint256).max ? _products.length : param.productId;
-        allowedPools[productId] = param.allowedPools;
+      if (param.allowedPools.length > 0) {
+        requirePoolExists(param.allowedPools);
```

Comment:
`requirePoolExists` does an external call for every invocation to get the pool count. That can be cached in a variable at the top of this function and the check can be inlined here.